docs/proposals/build-execution-using-build-run.md

```diff
@@ -0,0 +1,76 @@
+---
+title: build-execution-using-build-run
+authors:
+  - "@sbose78"
+  - "@zhangtbj"
+status: design
+---
+
+# Build execution using the BuildRun API
+
+
+## About
+
+A `BuildRun` is an immutable CR/object that represents a single execution of the `Build`.
+
+The simplest `BuildRun` looks like this:
+```
+apiVersion: build.dev/v1alpha1
+kind: BuildRun
+metadata:
+  generateName: account-service-build-
+spec:
+  buildRef:
+    name: kaniko-golang-build
+```
+
+The `BuildRun` API also provides a way to override/specify  execution time information. 
+
+As an example, a `Build` configuration should not have to accurately specify the resource requirements - the information on resource requirements is valuable at build execution time which varies from environment to environment.
+
+## Defining a BuildRun
+
+The `BuildRun`, apart from triggering a build execution, also provides the user an opportunity to specify/override a subset of Build configuration which are likely to change at build execution time.
+
+### What is deterministic
+
+#### What are we building
+* The source code that's being built into an image.
+* The application binary that's being packaged into an image.
+
+The above information wraps the integrity of the code
+that we are building into an image.
+
+#### How are we building
+* The `BuildStrategy` being used to build the image.
+* Inputs associated with the `BuildStrategy` : Dockerfile, builder image, etc.
+* The runtime base image being used to build a lean image.
+
+The above information wraps the know-how that drives the process of converting source code into an image.
+
+### What is non-deterministic
+
+* The Image being pushed to
+    * Repository
+    * Tag
+* The Service Account used to execute the build.
+* Execution-time resource requiements.
+
+As an example,
+the source code https://github.com/nodeshift-starters/nodejs-rest-http-crud might be pushed to different image repositories depending on who is executing the build. 
+
+### Deciding what should be overriden
```

Comment:
Hi Shoubhik,

Quick question, what this `### Deciding what should be overriden` block for?

should we answer here or we should confirm together?

I think the overriding rule is just covered between build and buildRun. is it also related with cluster and namespace?

Reply:
These are the questions we'll ask ourselves whenever we get into a dilemma about what should go into a BuildRun and what shouldn't.